Source/CarthageKit/Xcode.swift

```diff
@@ -787,7 +786,17 @@ public func buildDependencyProject(dependency: ProjectIdentifier, rootDirectoryU
 			}
 		}
 
-		if !NSFileManager.defaultManager().createSymbolicLinkAtURL(dependencyBinariesURL, withDestinationURL: rootBinariesURL, error: &error) {
+		// The relative path to this dependency's Carthage/Build folder, from
+		// the root.
+		let dependencyBinariesRelativePath = dependency.relativePath.stringByAppendingPathComponent(CarthageBinariesFolderPath)
+		let componentsForGettingTheHellOutOfThisRelativePath = Array(count: dependencyBinariesRelativePath.pathComponents.count - 1, repeatedValue: "..")
```

Comment:
I looked at `Repeat` before writing this code, and it didn't apply to my original conception. Of course, this final implementation would actually make perfect sense for it.

The benefit would be not needing to preallocate an array for the repeating components, but it's so small that it doesn't matter.